File: jlpt/native_ios/Sources/Domain/Models/Content.swift
import Foundation

struct Vocabulary: Identifiable, Equatable {
    let id: Int64
    let level: JlptLevel
    let word: String
    let reading: String
    let meaningVi: String
    let meaningEn: String?
    let partOfSpeech: String?
    let exampleSentence: String?
    let exampleReading: String?
    let exampleMeaning: String?
}

struct Kanji: Identifiable, Equatable {
    let id: Int64
    let level: JlptLevel
    let character: String
    let onyomi: String?
    let kunyomi: String?
    let meaningVi: String
    let strokeCount: Int?
    let exampleWords: String?
}

struct Grammar: Identifiable, Equatable {
    let id: Int64
    let level: JlptLevel
    let pattern: String
    let meaningVi: String
    let meaningEn: String?
    let usageNote: String?
    let exampleSentence: String?
    let exampleMeaning: String?
}

struct Passage: Identifiable, Equatable {
    let id: Int64
    let level: JlptLevel
    let title: String
    let content: String
}

struct ListeningAudio: Identifiable, Equatable {
    let id: Int64
    let level: JlptLevel
    let title: String
    let audioUrl: String
    let transcript: String?
    let durationSeconds: Int?
}

File: jlpt/native_android/app/src/main/java/com/jlpt/android/domain/model/Content.kt
package com.jlpt.android.domain.model

data class Vocabulary(
    val id: Long,
    val level: JlptLevel,
    val word: String,
    val reading: String,
    val meaningVi: String,
    val meaningEn: String?,
    val partOfSpeech: String?,
    val exampleSentence: String?,
    val exampleReading: String?,
    val exampleMeaning: String?
)

data class Kanji(
    val id: Long,
    val level: JlptLevel,
    val character: String,
    val onyomi: String?,
    val kunyomi: String?,
    val meaningVi: String,
    val strokeCount: Int?,
    val exampleWords: String?
)

data class Grammar(
    val id: Long,
    val level: JlptLevel,
    val pattern: String,
    val meaningVi: String,
    val meaningEn: String?,
    val usageNote: String?,
    val exampleSentence: String?,
    val exampleMeaning: String?
)

data class Passage(
    val id: Long,
    val level: JlptLevel,
    val title: String,
    val content: String
)

data class ListeningAudio(
    val id: Long,
    val level: JlptLevel,
    val title: String,
    val audioUrl: String,
    val transcript: String?,
    val durationSeconds: Int?
)
